internal/controller/monovertexrollout/monovertexrollout_controller.go

```diff
@@ -291,17 +291,27 @@ func (r *MonoVertexRolloutReconciler) processExistingMonoVertex(ctx context.Cont
 	}
 	switch inProgressStrategy {
 	case apiv1.UpgradeStrategyProgressive:
-		if mvNeedsToUpdate {
-			numaLogger.Debug("processing MonoVertex with Progressive")
-			done, err := progressive.ProcessResourceWithProgressive(ctx, monoVertexRollout, existingMonoVertexDef, r, r.client)
-			if err != nil {
-				return err
-			}
-			if done {
-				r.inProgressStrategyMgr.UnsetStrategy(ctx, monoVertexRollout)
+
+		// don't risk out-of-date cache while performing Progressive strategy - get
+		// the most current version of the MonoVertex just in case
+		existingMonoVertexDef, err = kubernetes.GetLiveResource(ctx, newMonoVertexDef, "monovertices")
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				numaLogger.WithValues("monoVertexDefinition", *existingMonoVertexDef).Warn("MonoVertex not found.")
+			} else {
+				return fmt.Errorf("error getting MonoVertex for status processing: %v", err)
 			}
 		}
 
+		numaLogger.Debug("processing MonoVertex with Progressive")
+		done, err := progressive.ProcessResourceWithProgressive(ctx, monoVertexRollout, existingMonoVertexDef, mvNeedsToUpdate, r, r.client)
+		if err != nil {
+			return err
+		}
+		if done {
+			r.inProgressStrategyMgr.UnsetStrategy(ctx, monoVertexRollout)
+		}
+
```

Comment:
just like with Pipelines, if we're in the middle of "progressive" strategy, get the latest version of the MVX just to be sure rather than going to cache (note this is getting the one labeled "promoted")